packages/fluent/scss/chip/_layout.scss

```diff
@@ -0,0 +1,198 @@
+        min-width: 0;
+        display: flex;
+        flex-flow: row nowrap;
+        align-items: center;
+        overflow: hidden;
+        flex: 1 1 auto;
+    }
+    .k-chip-content:first-child {
+        margin-inline-start: var( --kendo-chip-spacing, #{ $kendo-chip-spacing } );
+    }
+    .k-chip-content:last-child {
+        margin-inline-end: var( --kendo-chip-spacing, #{ $kendo-chip-spacing } );
+    }
+
+
+    // Chip text
+    .k-chip-text,
+    .k-chip-label {
+        white-space: nowrap;
+        text-overflow: ellipsis;
+        overflow: hidden;
+        flex: 1 1 auto;
+    }
+
+
+    // Chip avatar
+    .k-chip-avatar {
+        width: var( --INTERNAL-kendo-chip-avatar-width, #{ $kendo-chip-avatar-width });
+        height: var( --INTERNAL-kendo-chip-avatar-height, #{ $kendo-chip-avatar-height });
+        flex-basis: var( --INTERNAL-kendo-chip-avatar-flex-basis, #{ $kendo-chip-avatar-flex-basis });
+        border-radius: 50%;
+        background-size: cover;
+        background-position: center;
+        flex: none;
+    }
+
+
+    // Chip icon
+    .k-chip-icon {
+        font-size: inherit;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        flex: none;
+    }
+    .k-ie .k-chip-icon {
+        margin-right: var( --kendo-chip-spacing, #{ $kendo-chip-spacing } );
+    }
+
+
+    // Actions
+    .k-chip-actions {
+        margin: -5em 0;
+        flex: none;
+        display: flex;
+        flex-flow: row nowrap;
+        align-items: center;
+        align-self: center;
+    }
+    .k-chip-action {
+        flex: none;
+        display: flex;
+        flex-flow: row nowrap;
+        align-items: center;
+        align-self: center;
+    }
+
+
+    // Legacy chip icons
+    .k-remove-icon {
+        font-size: inherit;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        flex: none;
+    }
+    .k-ie .k-remove-icon {
+        margin-right: 0;
+        margin-left: var( --kendo-chip-spacing, #{ $kendo-chip-spacing } );
+    }
+
+
+    // Chip list
+    .k-chip-list {
+        min-width: 0px;
+        display: inline-flex;
+        flex-wrap: wrap;
+        align-items: center;
+        position: relative;
+    }
+
+
+    // RTL
+    .k-ie .k-chip.k-rtl,
+    .k-ie .k-rtl .k-chip,
+    .k-ie .k-chip[dir="rtl"] {
+        .k-chip-icon {
+            margin-right: 0;
+            margin-left: var( --kendo-chip-spacing, #{ $kendo-chip-spacing } );
+        }
+
+        .k-remove-icon {
+            margin-left: 0;
+            margin-right: var( --kendo-chip-spacing, #{ $kendo-chip-spacing } );
+        }
+    }
+
+
+
+    // Sizes
+    @each $size, $size-props in $kendo-chip-sizes {
+        $_padding-x: map-get( $size-props, padding-x );
+        $_padding-y: map-get( $size-props, padding-y );
+        $_font-size: map-get( $size-props, font-size );
+        $_line-height: map-get( $size-props, line-height );
+        $_avatar-size: 1em;
```

Comment:
The avatar is dependant on the chip size because the avatar needs to have a width / height container because it will take the sizes of the image / icon itself.

Reply:
The avatar is not dependent on the chip size because of two reasons:

1. Its size is hardcoded to `1em` everywhere in the code (in both the `layout` and the `variables` file).
2. If we want it to be dependent on the chip size, then (apart from un-hardcoding the size) the code should be nested in `.k-chip-#{$size}` like so:

```
        .k-chip-#{$size} {
            --INTERNAL-kendo-chip-padding-x: #{ $_padding-x };
            --INTERNAL-kendo-chip-padding-y: #{ $_padding-y };
            --INTERNAL-kendo-chip-font-size: #{ $_font-size };
            --INTERNAL-kendo-chip-line-height: #{ $_line-height };

            .k-chip-avatar {
                --INTERNAL-kendo-chip-avatar-width: #{ $_avatar-size };
                --INTERNAL-kendo-chip-avatar-height: #{ $_avatar-size };
                --INTERNAL-kendo-chip-avatar-flex-basis: #{ $_avatar-size };
            }
        }
```